cilantro_audit/completed_audits_list_page.py

```diff
@@ -0,0 +1,98 @@
+import kivy
+from kivy.lang import Builder
+from kivy.properties import ObjectProperty
+from kivy.uix.button import Button
+from kivy.uix.label import Label
+from kivy.uix.screenmanager import Screen
+from mongoengine import connect
+
+from cilantro_audit.completed_audit import CompletedAudit
+from cilantro_audit.constants import KIVY_REQUIRED_VERSION, PROD_DB, SEVERITY_PRECEDENCE
+
+kivy.require(KIVY_REQUIRED_VERSION)
+
+kvfile = Builder.load_file("./widgets/completed_audits_list_page.kv")
+
+connect(PROD_DB)
+
+
+class CompletedAuditsListPage(Screen):
+    date_col = ObjectProperty()
+    title_col = ObjectProperty()
+    audit_list = ObjectProperty()
+    auditor_col = ObjectProperty()
+    severity_col = ObjectProperty()
+    refresh_button = ObjectProperty()
+
+    def __init__(self, **kw):
+        super().__init__(**kw)
+        self.date_col.bind(minimum_height=self.audit_list.setter("height"))
+        self.title_col.bind(minimum_height=self.audit_list.setter("height"))
+        self.auditor_col.bind(minimum_height=self.audit_list.setter("height"))
+        self.severity_col.bind(minimum_height=self.audit_list.setter("height"))
+        self.audits = []
+        self.load_completed_audits()
+
+    """Sorts list items by title."""
+
+    def sort_by_title(self):
+        self.audits = sorted(self.audits, key=lambda obj: (
+        obj.title, SEVERITY_PRECEDENCE[obj.severity.severity], obj.datetime, obj.auditor))
+        self.refresh_completed_audits()
+
+    """Sorts list items by datetime."""
+
+    def sort_by_date(self):
+        self.audits = sorted(self.audits, key=lambda obj: (
+        obj.datetime, SEVERITY_PRECEDENCE[obj.severity.severity], obj.title, obj.auditor))
```

Comment:
I'm assuming adding the `reverse=True` parameter to `sorted` would resolve the backwards sorting we don't want right now

Reply:
I don't think it will, though. I've been reading through the documentation. 

The thing is we want `severity : ascending`, `datetime : descending`, `title : ascending`, `auditor : ascending`.

Reverse will reverse all of it, and I need it on a per-key basis. Not seeing an easy way to do that yet.

Reply:
SQL queries make it easy, but I don't think Python will let you do it per key like this.